### PR TITLE
Run JDK 21 workflows with 21.0.4.

### PR DESCRIPTION
**Pull request by gianm**
To work around #17429, run our JDK 21 workflows with version 21.0.4. It does not appear to have this problem.

### CODE REVIEW

**Review by cryptoe — APPROVED** [review]

File: .github/workflows/standard-its.yml [hunk 1]
```diff
@@ -94,7 +94,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jdk: [8, 17, 21]
+        jdk: ['8', '17', '21.0.4']
```
Comment (abhishekrb19) [inline .github/workflows/standard-its.yml:97]: May want to update this check as well with the more specific `21.0.4` JDK version - https://github.com/apache/druid/blob/master/integration-tests/script/docker_build_containers.sh#L29

**Review by abhishekrb19 — APPROVED** [review]

File: .github/workflows/standard-its.yml (hunk 1 above)
Comment (kgyrtkirk) [reply .github/workflows/standard-its.yml]: I've unpatched this file for now - wdyt?

Comment (gianm) [reply .github/workflows/standard-its.yml]: Sounds OK to me.

**Comment by gianm** [comment]
The JDK 21 unit tests passed. I've added some comments to link back to #17429 and pushed again. If they pass again, that's a good sign, and I'll merge this.

File: .github/workflows/standard-its.yml (hunk 1 above)
Comment (abhishekrb19) [reply .github/workflows/standard-its.yml]: Sounds reasonable to me

**Pull request merged**